app/js/enhancedDate.js

```diff
@@ -1,5 +1,72 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function() { 
+    'use strict';
+    var myDate = null;
+    var dateSet = false;     
+
+    return {    
+        getDate: function(returnObj) {  
+            if (!dateSet)  {
+                this.setDate();
+            }
+            if (returnObj) {
+                return myDate;
+            } else {
+                return myDate.getTime();
+            }
+        },
+        setDate: function(newDate) {
+            if (isNaN(Date.parse(newDate))) {
+                myDate = newDate();           
+            }
+            else {
+                myDate = new Date(newDate);
+                dateSet = true;                                     
+            }
+            myDate.setHours(0, 0, 0, 0);    
```

Comment:
This line is unnecessary and will change the time set when you don't want it to be.